micrometer-core/src/main/java/io/micrometer/core/instrument/binder/mongodb/DefaultMongoCommandTagsProvider.java

```diff
@@ -68,4 +79,41 @@ private String getAndRemoveCollectionNameForCommand(CommandEvent event) {
         String collectionName = inFlightCommandCollectionNames.remove(event.getRequestId());
         return collectionName != null ? collectionName : "unknown";
     }
+
+    /**
+     * Attempts to determine the name of the collection a command is operating on.
+     *
+     * <p>Because some commands either do not have collection info or it is problematic to determine the collection info,
+     * there is an allow list of command names {@link #COMMANDS_WITH_COLLECTION_NAME} used. If {@code commandName} is
+     * not in the allow list or there is no collection info in {@code command}, it will use the content of the
+     * {@code 'collection'} field on {@code command}, if it exists.
+     *
+     * <p>Taken from <a href="https://github.com/openzipkin/brave/blob/master/instrumentation/mongodb/src/main/java/brave/mongodb/TraceMongoCommandListener.java#L115>TraceMongoCommandListener.java</a>
+     *
+     * @param commandName name of the mongo command
+     * @param command mongo command object
+     * @return optional collection name or empty if could not be determined or not in the allow list of command names
+     */
+    protected Optional<String> determineCollectionName(String commandName, BsonDocument command) {
```

Comment:
This is not private for 2 reasons.

1. VisibleForTesting - I still want to test this piece of the tags provider solo (and re-use the existing MongoCommandUtil tests).  Typically I would do that w/ package protected modifier. I went w/ `protected` though for the following reason. 

2. If a consumer needs another way to determine the collection name from a command event they could extend the DTP and override this method. I don't want to encourage that but it would give a consumer the ability to adjust the behavior if need be. @jonatan-ivanov what are your stance on "consumer-friendly and leaving some key methods extendable" vs. "lock it all down as any public or protected API is an increase in the dependency blast radius that we have to support". ?

Reply:
👍🏼 
I think it is great! It provides an escape route, also one of the original goal of the tag provider was making it flexible enough so that users can override the original/provide their own, this matches with that nicely.